src/receive_imf.rs

```diff
@@ -1746,7 +1749,16 @@ RETURNING id
         // delete it.
         needs_delete_job = true;
     }
-
+    if restore_protection {
+        chat_id
+            .set_protection(
+                context,
+                ProtectionStatus::Protected,
+                mime_parser.timestamp_sent,
```

Comment:
Using `timestamp_sent` here doesn't look correct, it rather should be `timestamp_rcvd`. `timestamp_sent` is a `Date:` of the received old message

Reply:
Actually this doesn't matter as protection messages are sorted to bottom anyway, but i changed this to `timestamp_rcvd`